src/SOS/Strike/strike.cpp

```diff
@@ -2464,7 +2464,8 @@ size_t FormatGeneratedException (DWORD_PTR dataPtr,
             WCHAR filename[MAX_LONGPATH] = W("");
             ULONG linenum = 0;
             if (bLineNumbers && 
-                SUCCEEDED(GetLineByOffset(TO_CDADDR(ste.ip), &linenum, filename, _countof(filename))))
+                // Adjust the ip for leaf node frames
```

Comment:
I would expand this comment greatly to describe:
a) Why we are returning line numbers that do not match the IP we are reporting? (I am guessing it is because we'd actually prefer to show the IP of the faulting instruction or the call which faulted, but it is complicated to compute that IP given only the IP of the following instruction so we don't bother. However for the line number we do have an easy hack of subtracting 1 from the IP and then mapping it)
b) Why only change the line number of the leaf and not the line number of the callers? I am guessing that all the caller frames will show the line for the assembly instruction that is *after* the call, but the leaf frame shows the line which threw the exception, not the one which follows it?
c) What value do we expect ste.ip to have here? This is my guess, but the comment ideally would answer it definitively: ste.ip is the IP of the instruction that follows in memory the one which faulted. For software exceptions or non-leaf frames the faulting instruction is a call instruction whose implementation elsewhere indirectly raised the exception. For the leaf frame of hardware exceptions the faulting instruction is the instruction which caused the CPU to trap into the kernel.